compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2212,16 +2212,17 @@ impl<'tcx> TyCtxt<'tcx> {
         if ts.is_empty() { List::empty() } else { self._intern_bound_variable_kinds(ts) }
     }
 
-    pub fn mk_fn_sig<I>(
+    pub fn mk_fn_sig<I, T>(
         self,
         inputs: I,
         output: I::Item,
         c_variadic: bool,
         unsafety: hir::Unsafety,
         abi: abi::Abi,
-    ) -> <I::Item as InternIteratorElement<Ty<'tcx>, ty::FnSig<'tcx>>>::Output
+    ) -> T::Output
     where
-        I: Iterator<Item: InternIteratorElement<Ty<'tcx>, ty::FnSig<'tcx>>>,
+        I: Iterator<Item = T>,
```

Comment:
Side-note, I wonder if this (and similar) should take `IntoIterator` for `I`, then we can avoid the `.into_iter()` calls later?

Reply:
For `mk_fn_sig` it's a great idea, precisely because there isn't an `intern_fn_sig` that can take a slice. For the other `mk_foo` functions that take an iterator, it's not much use because `intern_foo` exists.